toolkit/tools/graphpkgfetcher/graphpkgfetcher.go

```diff
@@ -168,27 +169,73 @@ func resolveSingleNode(cloner *rpmrepocloner.RpmRepoCloner, node *pkggraph.PkgNo
 		return
 	}
 
-	if !fetchedPackages[resolvedPackage] {
-		desiredPackage := &pkgjson.PackageVer{
-			Name: resolvedPackage,
+	for _, resolvedPackage := range resolvedPackages {
+		if !fetchedPackages[resolvedPackage] {
+			desiredPackage := &pkgjson.PackageVer{
+				Name: resolvedPackage,
+			}
+
+			err = cloner.Clone(cloneDeps, desiredPackage)
+			if err != nil {
+				logger.Log.Errorf("Failed to clone '%s' from RPM repo. Error: %s", resolvedPackage, err)
+				return
+			}
+			fetchedPackages[resolvedPackage] = true
+
+			logger.Log.Debugf("Fetched '%s' as potential candidate.", resolvedPackage)
 		}
+	}
+
+	err = assignRPMPath(node, outDir, resolvedPackages)
+	if err != nil {
+		logger.Log.Errorf("Failed to find an RPM to provide '%s'. Error: %s", node.VersionedPkg.Name, err)
+		return
+	}
+
+	node.State = pkggraph.StateCached
+
+	logger.Log.Infof("Choosing '%s' to provide '%s'.", filepath.Base(node.RpmPath), node.VersionedPkg.Name)
+
+	return
+}
 
-		err = cloner.Clone(cloneDeps, desiredPackage)
+func assignRPMPath(node *pkggraph.PkgNode, outDir string, resolvedPackages []string) (err error) {
+	rpmPaths := []string{}
+	for _, resolvedPackage := range resolvedPackages {
+		rpmPaths = append(rpmPaths, rpmPackageToRPMPath(resolvedPackage, outDir))
+	}
+
+	node.RpmPath = rpmPaths[0]
```

Comment:
I think we should have a check that `rpmPaths` does at least have 1 entry.

Reply:
Good catch.